lib/mix/tasks/scenic.gen.component.ex

```diff
@@ -0,0 +1,74 @@
+defmodule Mix.Tasks.Scenic.Gen.Component do
+  @shortdoc "Creates scaffolding for a new Scenic component."
+
+  @moduledoc """
+  Generates Scene boilerplate for a new component.
+
+  ```bash
+  mix scenic.gen.component component_name
+  ```
+  """
+  use Mix.Task
+
+  import Mix.Generator
+  alias ScenicNew.Common
+
+  @switches [
+    app: :string,
+    module: :string
+  ]
+
+
+  # --------------------------------------------------------
+  @doc false
+  def run(argv) do
+    if Mix.Project.umbrella?() do
+      Mix.raise "mix scenic.gen.component must be invoked from within your scenic application root directory."
```

Comment:
`Elixir`'s formatter wants to break it as this is too long. Since we are using `scenic.gen.component` string part at least 2 times in this file I suggest to make it an attribute and use it as:
```elixir
@task "scenic.gen.component"
"#{@task}"
```